### PR TITLE
Sign specific

### PR DESCRIPTION
**Pull request by Rlamboll**
# Pull request

Please confirm that this pull request has done the following:

- [x] Tests added
- [x] Documentation added (where applicable)
- [x] Example added (either to an existing notebook or as a new notebook, where applicable)
- [x] Description in ``CHANGELOG.rst`` added

## Adding to CHANGELOG.rst

Please add a single line in the changelog notes similar to one of the following:

```
- (`#XX <https://github.com/znicholls/silicone/pull/XX>`_) Added feature which does something
- (`#XX <https://github.com/znicholls/silicone/pull/XX>`_) Fixed bug identified in (`#YY <https://github.com/znicholls/silicone/issues/YY>`_)
```

### CODE REVIEW

**Comment by jkikstra** [comment]
A small note before I review the full PR later; in general I prefer to have notebooks be uploaded to the repository with cleared output cells (so not run), because this makes it easier to track changes in the code in the cell between versions, and it keeps the uploaded lines of code to a minimum.

**Comment by znicholls** [comment]
> A small note before I review the full PR later; in general I prefer to have notebooks be uploaded to the repository with cleared output cells (so not run), because this makes it easier to track changes in the code in the cell between versions, and it keeps the uploaded lines of code to a minimum.

The one problem with this is that anyone who looks at the repo has to download and run the notebooks before they can see what the outputs are. Using something like this (https://www.reviewnb.com/) or nbdiff might make it way easier.

File: src/silicone/multiple_infillers/decompose_collection_with_time_dep_ratio.py [hunk 1]
```diff
@@ -73,6 +73,19 @@ def _construct_consistent_values(self, aggregate_name, components, db_to_generat
...
+        Parameters
+        ----------
+        df : obj:`pyam.IamDataFrame`
+            The dataframe whose units we want
+
+        Returns
+        -------
+        Set
+            The set of units
+        """
```
Comment (jkikstra) [inline src/silicone/multiple_infillers/decompose_collection_with_time_dep_ratio.py]: Could you make it more explicit what kind of set of units is returned?

File: tests/integration/simple_infillers/test_decompose_collection_with_time_dep_ratio.py [hunk 2]
```diff
@@ -172,7 +173,7 @@ def test_db_error_preexisting_follow_data(self, test_db):
             tcruncher.infill_components(variable_follower, variable_leaders, test_db)
 
     def test_construct_consistent_error_multiple_units(self, test_db):
-        # test that crunching fails if there's no data about the follower gas in the
+        # test that crunching fails if there's no bad units in the
```
Comment (jkikstra) [inline tests/integration/simple_infillers/test_decompose_collection_with_time_dep_ratio.py]: This line of description I do not understand

File: tests/integration/simple_infillers/test_decompose_collection_with_time_dep_ratio.py [hunk 3]
```diff
@@ -184,7 +185,9 @@ def test_construct_consistent_error_multiple_units(self, test_db):
         with pytest.raises(ValueError, match=error_msg):
             tcruncher._construct_consistent_values(aggregate_name, components, test_db)
 
+
     def test_relationship_usage_not_enough_time(self, test_db, test_downscale_df):
```
Comment (jkikstra) [inline tests/integration/simple_infillers/test_decompose_collection_with_time_dep_ratio.py]: It's great that you have so many tests. If you would want and have time; a line of description per test would be nice.

Comment (Rlamboll) [reply tests/integration/simple_infillers/test_decompose_collection_with_time_dep_ratio.py]: Yeah, I'll do that, they're not always obvious

File: tests/integration/simple_infillers/test_decompose_collection_with_time_dep_ratio.py (hunk 2 above)
Comment (Rlamboll) [reply tests/integration/simple_infillers/test_decompose_collection_with_time_dep_ratio.py]: ok, reworded

File: src/silicone/multiple_infillers/decompose_collection_with_time_dep_ratio.py (hunk 1 above)
Comment (Rlamboll) [reply src/silicone/multiple_infillers/decompose_collection_with_time_dep_ratio.py]: I'll write more for this anyway

**Pull request merged**